cores/esp32/IPAddress.cpp

```diff
@@ -20,68 +20,86 @@
 #include <Arduino.h>
 #include <IPAddress.h>
 #include <Print.h>
+#include <StreamString.h>
 
-IPAddress::IPAddress()
-{
-    _address.dword = 0;
+IPAddress::IPAddress() {
+#if LWIP_IPV6
+  _ip = *IP6_ADDR_ANY;
+#else
+  _ip = *IP_ADDR_ANY;
+#endif
```

Comment:
Default constructor does not set all bytes to 0.
Thus when setting an IPv4 address only using `operator[]`, the IPv6 bytes will be uninitialized and thus any compare (`operator==` not using an `uint8_t` pointer) will fail.